timemachine/fe/topology.py

```diff
@@ -319,6 +319,74 @@ def parameterize_nonbonded(self, ff_q_params, ff_lj_params):
         return combined_qlj_params, interpolated_potential
 
 
+class RelativeFreeEnergyForcefield(BaseTopology):
+    """
+    Used to run the same molecule under different forcefields.
+    Currently only changing the nonbonded parameters is supported.
+
+    The parameterize_* methods should be passed a list of parameters
+    corresponding to forcefield0 and forcefield1.
+    """
+
+    def __init__(self, mol, forcefield0, forcefield1):
+        """
+        Utility for working with a single ligand.
+
+        Parameter
+        ---------
+        mol: ROMol
+            Ligand to be parameterized
+
+        forcefield0: ff.Forcefield
+            A convenience wrapper for forcefield lists.
+
+        forcefield1: ff.Forcefield
+            A convenience wrapper for forcefield lists.
+
+        """
+        self.mol = mol
+        self.ff = forcefield0
+        self.ff1 = forcefield1
+        self.bt1 = BaseTopology(mol, forcefield1)
+
+    def parameterize_nonbonded(self, ff_q_params, ff_lj_params):
+        assert self.ff.lj_handle.smirks == self.ff1.lj_handle.smirks, "changing lj smirks is not supported"
+        assert self.ff.q_handle.smirks == self.ff1.q_handle.smirks, "changing charge smirks is not supported"
+        src_qlj_params, nb_potential = super().parameterize_nonbonded(ff_q_params[0], ff_lj_params[0])
+        dst_qlj_params, _ = self.bt1.parameterize_nonbonded(ff_q_params[1], ff_lj_params[1])
+
+        combined_qlj_params = jnp.concatenate([src_qlj_params, dst_qlj_params])
+        lambda_plane_idxs = np.zeros(self.mol.GetNumAtoms(), dtype=np.int32)
+        lambda_offset_idxs = np.zeros(self.mol.GetNumAtoms(), dtype=np.int32)
+
+        interpolated_potential = nb_potential.interpolate()
+        interpolated_potential.set_lambda_plane_idxs(lambda_plane_idxs)
+        interpolated_potential.set_lambda_offset_idxs(lambda_offset_idxs)
```

Comment:
Not 100% sure about this step -- is there a situation where we would want to simultaneously control the decoupling _and_ the parameter interpolation?

`u(x=x, params=interpolate(params_0, params_1, 0.33), lam=0.25)`

(Maybe not: probably suffices to interpolate params at decoupling distance = 0, and construct two classes corresponding to coupled vs. decoupled...)

Reply:
Yeah, probably not. At least I can't think of a reason to do this right now.